src/merkle.rs

```diff
@@ -751,16 +859,70 @@ impl Element for [u8; 32] {
     }
 }
 
-// This method returns the actual merkle tree length, but requires
-// that leafs is a power of 2.
-pub fn get_merkle_tree_len(leafs: usize) -> usize {
-    2 * next_pow2(leafs) - 1
+// Tree length calculation given the number of leafs in the tree and the branches.
+pub fn get_merkle_tree_len(leafs: usize, branches: usize) -> usize {
```

Comment:
if we already have `branches` as typenum, lets use that on these functions instead

Reply:
I can, but I was intentionally not since we need a usize there eventually, and I was trying not to 'infect' the store with requiring the typenums.

Reply:
actually, I am not sure. undecided

Reply:
Ok, I'll re-review that while updating.